### PR TITLE
#163388255 User should be able to rate an article 

### PR DESCRIPTION
**Pull request by bryan-munene**
#### What does this PR do?

This PR adds the rating functionality to the app. Users can rate an article only if they are authenticated. But the owner of the article cannot rate their own article.

#### Description of Task to be completed?

- Write Unit tests
- Add POST ratings to an article
- Add GET ratings to an article

#### How should this be manually tested?

- Clone the repo.
- git checkout `ft-rate-an-article-163388255`

- Make the following requests to _postman_:

| Request         | URL                | Description |
| ------------- | ------------- |------------------|
| `POST`  | `{BASE_URL}/api/articles/ <slug>/ratings` |Rate an article|
| `GET`  | `{BASE_URL}/api/articles/ <slug>/ratings`  |Fetch all ratings on an article |

#### Any background context you want to provide?

- The rating can be changed by overwriting the existing rating but cannot be deleted.
- An article owner cannot rate their own article
- Unauthenticated users cannot post ratings

#### What are the relevant pivotal tracker stories?
[#163388255](https://www.pivotaltracker.com/story/show/163388255)
#### Screenshots (if appropriate):

screen shot 2019-02-14 at 17 38 20
screen shot 2019-02-14 at 17 38 48

#### Questions:

None

### CODE REVIEW

File: authors/apps/profiles/tests/base_test.py [hunk 1]
```diff
@@ -69,5 +69,6 @@ def edit_profile(self, username):
     def authorize_user(self, user_login_details):
         # register a user
         self.register_user(data=self.user)
-        payload = self.login_a_user(data=user_login_details)
+        payload = self.login_user(data=user_login_details)
+        print(">>>>>>>>>>>>", payload)
```
Comment (sam-karis) [inline authors/apps/profiles/tests/base_test.py]: Remove the print statement

File: authors/apps/profiles/tests/test_profile.py [hunk 2]
```diff
@@ -26,6 +26,7 @@ def test_get_user_profile(self):
     def test_view_all_users_profiles(self):
         """Test that a user can view all user profiles"""
         response = self.register_user(data=self.user)
+        print(">>>>>>>>>HERE<<<<<<<<<", response)
```
Comment (sam-karis) [inline authors/apps/profiles/tests/test_profile.py]: Remove the print statement

**Review by sam-karis — CHANGES_REQUESTED** [review]

**Review by TooColline — COMMENTED** [review]
Nice work @bryan-munene works as expected 
image

File: authors/apps/profiles/tests/base_test.py (hunk 1 above)
Comment (bryan-munene) [reply authors/apps/profiles/tests/base_test.py]: done

File: authors/apps/profiles/tests/test_profile.py (hunk 2 above)
Comment (bryan-munene) [reply authors/apps/profiles/tests/test_profile.py]: done

File: authors/apps/rating/tests/test_rating.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+
+class TestRating(TestConfiguration):
+
+    def test_get_ratings_anonymous(self):
+        """
+        Test that a non-authenticated user can view an article's ratings
+        """
+        response = self.get_ratings(url=self.ratings_url)
+        # self.assertIn('ratings', response.data)
```
Comment (sam-karis) [inline authors/apps/rating/tests/test_rating.py]: Remove the commented code

File: authors/apps/rating/tests/test_rating.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        self.assertEqual(response.status_code, 200)
+
+    def test_get_ratings_authenticated(self):
+        """
+        Test that an authenticated user can view an article's ratings
+        """
+        response = self.get_ratings(
+            is_authenticated=True, url=self.ratings_url)
+        # self.assertIn('ratings', response.data)
```
Comment (sam-karis) [inline authors/apps/rating/tests/test_rating.py]: Remove the commented code

**Review by sam-karis — CHANGES_REQUESTED** [review]

File: authors/apps/rating/tests/test_rating.py (hunk 3 above)
Comment (bryan-munene) [reply authors/apps/rating/tests/test_rating.py]: done

File: authors/apps/rating/tests/test_rating.py (hunk 4 above)
Comment (bryan-munene) [reply authors/apps/rating/tests/test_rating.py]: done

**Review by calebrotich — APPROVED** [review]

**Review by sam-karis — APPROVED** [review]
LGTM

**Pull request merged**